src/Illuminate/Database/Eloquent/Model.php

```diff
@@ -1184,6 +1184,17 @@ public function getRouteKey()
     }
 
     /**
+     * Resolve route key value for retrieving a record.
+     *
+     * @param $keytoresolve Record identifier to resolve
```

Comment:
It's mixed  I believe - could be a string in the case of a Hashid or integer in the case of a raw numeric ID.